dashboard/src/utils/constructDate.js

```diff
@@ -1,5 +1,3 @@
-export const constructUTCDate = (strDate) => {
-  const date = new Date(strDate);
-  date.setUTCMinutes(date.getUTCMinutes() - date.getTimezoneOffset());
-  return date;
+export const dateFromUTCString = (strDate) => {
+  return new Date(strDate + "Z");
```

Comment:
Is this called anywhere other than converting the server's creation timestamp? At this point all the server timestamps should be full UTC ISO time.

One exception is the `server.deletion` metadata, which is a fixed string generated by `PUT`; old datasets will have bare "YYYY-MM-DD" strings, but new datasets should get a full ISO timestamp:

```
  {
    "controller": "dhcp31-136.perf.lab.eng.bos.redhat.com",
    "metadata": {
      "dataset.access": "private",
      "dataset.created": "2022-05-04T20:02:09+00:00",
      "dataset.owner": "david.butenhof@redhat.com",
      "dataset.uploaded": "2022-05-09T19:12:17.358371+00:00",
      "server.deletion": "2024-05-08T19:12:17.358371+00:00"
    },
    "name": "pbench-user-benchmark_single_2022.05.04T20.02.09",
    "run_id": "ec547b14082909974a0e2157fd706d77"
  },
```

Reply:
This is called at couple of places while selecting dates from date picker.

Reply:
Yeah, this code should engage with its input a little more -- e.g., if the input already has a timezone designator, then it obviously doesn't need to add a `Z`, but if the indicated time zone is anything other than UTC then we need to do some more significant manipulations than just creating a `Date` from it...but that stuff can come in another PR.

For now, if the user uses the GUI elements rather than typing in a date manually, what we seem to get from the `DatePicker` is a string in `YYYY-MM-DD` format, and adding a trailing `Z` will ensure that the `Date` c'tor treats it properly.